### DISCUSSION TITLE
Editing Overlays in QC report - dwi threshold used for extraction & number slices shown for MELODIC

### DISCUSSION THREAD

**Discussion by dsodums** [McIntosh-Lab/tvb-ukbb#161] [General]
I was wondering if its simple to change the QC code for the dwi extraction default threshold used when displaying the B0 image underlay as when using the Rockland dataset with dwi bval=1500 the image is really faint.

Also, is it possible to change how many slices are shown for the MELODIC report?

Thanks,
Devin

**Comment by justinliuyuwang** [comment]
We are still in the process of fleshing out the [Modifying QC Generation](https://github.com/McIntosh-Lab/tvb-ukbb/wiki/Modifying-QC-Generation) page, but you can find some details for modifying the B0 underlay there. 

Currently, calls to `lightbox.sh` from `tvb-ukbb/tvb_bb_QC/tvb_bb_QC.sh`  generate the dwi extraction B0 image overlays. The -b and -c options correspond to the brightness and contrast of the underlay image in the final overlay image. Please note that there are two calls to `lightbox.sh` for the extraction analysis (one unmasked and one masked). You will just need to fiddle with the -b and -c values for those two calls to `lightbox.sh` until you find a setting that works for your dataset.

[Unmasked](https://github.com/McIntosh-Lab/tvb-ukbb/blob/9b89811e3c1acde1ecc73df9ff668bbdc6533d1b/tvb_bb_QC/tvb_bb_QC.sh#L189)
```
#dMRI extraction unmasked
		$BB_BIN_DIR/tvb_bb_QC/lightbox.sh -b 62.718401747457314 -c 96.08735171986915 /dMRI/dMRI/data_B0.nii.gz $dirSubject dMRI_extraction_unmasked
```

[Masked](https://github.com/McIntosh-Lab/tvb-ukbb/blob/9b89811e3c1acde1ecc73df9ff668bbdc6533d1b/tvb_bb_QC/tvb_bb_QC.sh#L196)
```
	#dMRI extraction masked
		$BB_BIN_DIR/tvb_bb_QC/lightbox.sh -o /dMRI/dMRI/nodif_brain_mask.nii.gz -b 62.718401747457314 -c 96.08735171986915 /dMRI/dMRI/data_B0.nii.gz $dirSubject dMRI_extraction_masked
```

Unfortunately, the images shown in the MELODIC and EDDY_QUAD reports are generated by their respective programs. The workaround would be to add calls to `tvb-ukbb/tvb_bb_QC/lightbox.sh` and/or `tvb-ukbb/tvb_bb_QC/edges.sh` into `tvb-ukbb/tvb_bb_QC/tvb_bb_QC.sh` so that the QC sub-pipeline will generate the desired slices. These new images would not appear on the QC Report, however; they would only appear as png files in the `images` directory of the subject's QC folder. We are working on a way for users to have their new QC images included into the html report.

**Reply by dsodums** [reply]
Thank you! Apologies that I didn't see the QC modifying page before, that's very helpful and should give me what I need to play around with the threshold values for display.

Understood regarding the MELODIC outputs, thanks again for the help :)

**Reply by justinliuyuwang** [reply]
No problem and no worries! Those pages are there for us to point to when users have questions :)

**Comment by dsodums** [comment]
I'm actually a bit confused, I went to edit those two lines in tvb-ukbb/tvb_bb_QC/tvb_bb_QC.sh  but they look very different in my version of the file than what you had pasted:

 #dMRI extraction unmasked
                $BB_BIN_DIR/tvb_bb_QC/lightbox.sh /dMRI/dMRI/data_B0.nii.gz $dirSubject dMRI_extraction_unmasked


                #$BB_BIN_DIR/tvb_bb_QC/image_gen_link.sh $dirSubject "uw_under" "../../dMRI/dMRI/data_B0.nii.gz" "NA22" "NA22_link" "NA23" "NA23_link" 0


        #dMRI extraction masked
                $BB_BIN_DIR/tvb_bb_QC/lightbox.sh -o /dMRI/dMRI/nodif_brain_mask.nii.gz /dMRI/dMRI/data_B0.nii.gz $dirSubject dMRI_extraction_masked


                $BB_BIN_DIR/tvb_bb_QC/image_gen_link.sh $dirSubject "dex_under" "../../dMRI/dMRI/data_B0.nii.gz" "dex_over1" "../../dMRI/dMRI/nodif_brain_mask.nii.gz" "NA24" "NA24_link" 0

Am I missing something?

**Reply by justinliuyuwang** [reply]
Sorry, the two lines to be changed are in lines [190](https://github.com/McIntosh-Lab/tvb-ukbb/blob/9b89811e3c1acde1ecc73df9ff668bbdc6533d1b/tvb_bb_QC/tvb_bb_QC.sh#L190) (unmasked) and line [197](https://github.com/McIntosh-Lab/tvb-ukbb/blob/9b89811e3c1acde1ecc73df9ff668bbdc6533d1b/tvb_bb_QC/tvb_bb_QC.sh#L197) (masked). The main branch does not have brightness and contrast options for this analysis. When you add these options, they should be placed right after the `$BB_BIN_DIR/tvb_bb_QC/lightbox.sh` in the call, like what I had pasted in my previous post.

**Comment by dsodums** [comment]
Oh perfect, thank you so much!

I'm hoping to test just the QC stage using an interactive session on Compute Canada rather than submitting the entire pipeline. I have that session running but what would be the easiest way to test just this stage? I've played around with pasting just that part of the code but think I'm missing something so any pointers would be greatly appreciated.

**Reply by justinliuyuwang** [reply]
In [tvb-ukbb/bb_pipeline_tools/bb_pipeline.py](https://github.com/McIntosh-Lab/tvb-ukbb/blob/main/bb_pipeline_tools/bb_pipeline.py) I would comment out calls to other subpipes, i.e. remove the following lines (you may have to look around the file for them, they are not on consecutive lines):

```
bb_pipeline_struct(subject, runTopup, fileConfig)
bb_pipeline_func(subject, fileConfig)
bb_pipeline_diff(subject, fileConfig)
bb_IDP(
        subject, fileConfig
    )
```


You will then be able to run the pipeline as normal and it will only run the QC subpipeline.

**Reply by dsodums** [reply]
Thank you so much!!
